content/blog/intro-to-gamedev/index.md

```diff
@@ -0,0 +1,560 @@
+
+Let's dive in!
+
+## Scope
+
+It's very easy to overshoot when scoping any piece of software, especially when you can clearly see many ways in which your software can be expanded which is the case for pretty much any kind of game (power-ups, sound effects, types of entities, etc.). In these kinds of contexts, (over-)engineering the code to be prepared for future implementations seems like the correct way to go.
+
+In this case, however, we have a pretty clear idea of what we want our game to look like: It's a simple infinite runner, where a couple of obstacles spawn from time to time and the score is directly based on how much time we can succesfully avoid them.
+
+It will be hard, but let's try to keep the code and program simple and pragmatic. There's value in making the code work more systematically but we are looking into low-friction implementation since we want this to work as an introduction to a couple of ideas and technologies. Furthermore, since we are learning some of these technologies, I'll try to be as explicit as possible while exploring some of their defining features (in general though, I would recommend diving deeper to each of the concepts).
+
+## Tech requirements
+
+Keeping the scope in mind, let's try to deduce and compress what we need from a technology perspective:
+
+- We want the game to run on the browser, which pretty much inevitably means we are going to use either Javascript, or something that ends up turning into Javascript.
+- We have to do game-specific tasks such as render images and process input. Luckily the browser gives us good support for both of them through things like WebGL and Javascript's input system respectively.
+- We would also like it to be as responsive as possible since it makes for a better user experience.
+
+Note that we don't really need brazing fast graphic drawing for this since the game will not render thousands of objects at once, so using WebGL is probably overkill. However, it's a good way to get into the topic and so we'll take the chance to do it that way. WebGL is fairly complicated to work with from scratch, so after doing a bit of research I have decided we can use PixiJS to render stuff a pretty simple set of Javascript libraries for a 2D renderer that runs over WebGL (which in turn means we are using hardware acceleration to render stuff). We can also go a step further and try to use it with Typescript, to get that sweet, sweet type-checking.
+
+## Starting out
+
+In the world of web development, it's common
+We are going to use a starter template, which will solve some of these issues for us. Specifically, the starter is the following: https://github.com/llorenspujol/parcel-pixijs-quickstarter. It includes Parcel, PixiJS and TypeScript. PixiJS started shipping with Type information since version 5, which is the one being included in the starter template. This means we don't have to import extra packages to be able to type-check PixiJS code.
```

Comment:
I guess one of the two lines is the right one? `In the world of web development, it's common` or `We are going`, I vote for the latter